src/pages/Board/Board.tsx

```diff
@@ -86,20 +88,11 @@ export default function Board(): JSX.Element {
           placeholder="One more board..."
           onChange={(event): void => {
             if (!isStringValid(event.currentTarget.value))
-              setMistake({
-                text: mistake.text,
-                show: mistake.show,
-                firstShow: false,
-              });
-            else
-              setMistake({
-                text: mistake.text,
-                show: false,
-                firstShow: false,
-              });
+              setMistake((prevState) => ({ ...prevState, firstShow: false }));
```

Comment:
```suggestion
             const newMistakeState = { 
              firstShow: false,
              show: isStringValid(event.currentTarget.value), /// эту строчку нужно подумать, если у вас show тру стоит, то такая запись норм должна зайти
             }
             setMistake((prevState) => ({ ...prevState,  ...newMistakeState }));
```
и в итоге мы убираем иф и оставляем один сетер в стейт, должно быть красивенько =)